### PR TITLE
Lesson01 Assignment Files

### PR DESCRIPTION
**Pull request by KyleCreek**
Completed Assignments have been added to my Git

### CODE REVIEW

**Comment by KyleCreek** [comment]
Adding Assignment02 Homework.

File: students/KyleCreek/lesson01/assignment/test_integration.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
+"""
+# Title: test_integration.py
+# Desc: tests the integration of Main.py
+# Change Log: (Who, When, What)
+# KCreek, 4/14/2019, Created Script
```
Comment (majstorpoki) [inline students/KyleCreek/lesson01/assignment/test_integration.py:5]: I like this documentation

File: students/KyleCreek/lesson01/assignment/test_integration.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        """
+        # Create patch to handle each instance of print
+        with unittest.mock.patch('sys.stdout', new=io.StringIO()) as fake_out:
+            Main.get_price()
+
+        # Obtain StringIO output from get price
+        fake_out = fake_out.getvalue()
+        fake_out = fake_out.strip()
+        # Verify Main.get_price() produces desired output
+        self.assertEqual(fake_out, "Get price")
```
Comment (majstorpoki) [inline students/KyleCreek/lesson01/assignment/test_integration.py:67]: It checks whether the program printed "Get price"

File: students/KyleCreek/lesson01/assignment/test_integration.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        # Create a list of inputs as 'side_effects' to handle
+        # each instances of 'input()' function in Main.py
+        mock_input1 = [1, "Couch", 100, "y", "Plastic", "L"]
+
+        # Create to handle input function of Main.py and pass desired
+        # Inputs
+        with unittest.mock.patch('builtins.input', side_effect=mock_input1):
+            # Create a new item instance based on mock_input1 inputs
+            Main.add_new_item()
+            self.assertIn(1, Main.FULL_INVENTORY)
```
Comment (majstorpoki) [inline students/KyleCreek/lesson01/assignment/test_integration.py:85]: What does this check?

File: students/KyleCreek/lesson01/assignment/test_integration.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        # Create a list of inputs as 'side_effects' to handle
+        # each instances of 'input()' function in Main.py
+        mock_input2 = [2, "Blender", 100, "n", 'y', "Sony", 100]
+
+        # Create to handle input function of Main.py and pass desired
+        # Inputs
+        with unittest.mock.patch('builtins.input', side_effect=mock_input2):
+            # Create a new item instance based on mock_input2 inputs
+            Main.add_new_item()
+            self.assertIn(2, Main.FULL_INVENTORY)
```
Comment (majstorpoki) [inline students/KyleCreek/lesson01/assignment/test_integration.py:103]: The idea of your test is: 1) Create a record to be inserted; 2) Mock input from the record (list); 3) Check whether the record with particular ID is in FULL_INVENTORY; 3) Check whether the values are completely inserted

File: students/KyleCreek/lesson01/assignment/test_integration.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+            # Create an instance of a new item
+            Main.add_new_item()
+            # Create a patch to handle each instance of 'print()'
+            with unittest.mock.patch('sys.stdout', new=io.StringIO()) \
+                    as fake_out:
+                # Call item_info() function and verify output
+                Main.item_info()
+                fake_out = fake_out.getvalue()
+                fake_out = fake_out.strip()
+                self.assertEqual("Item not found in inventory", fake_out)
```
Comment (majstorpoki) [inline students/KyleCreek/lesson01/assignment/test_integration.py:150]: Nice

File: students/KyleCreek/lesson01/assignment/test_unit.py [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+        class is instantiated.
+        :return: None
+
+        """
+        # Create an instance of the Inventory Class
+        inventory = Inventory("1", "2", "3", "4")
+
+        # Test to verify data has been passed through the
+        # initializer
+        self.assertIsNotNone(inventory.product_code)
```
Comment (majstorpoki) [inline students/KyleCreek/lesson01/assignment/test_unit.py:33]: Each component initialized

File: students/KyleCreek/lesson01/assignment/test_unit.py [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+        """
+        # Create an instance of the Inventory Class
+        inventory = Inventory("1", "2", "3", "4")
+        # Obtain dictionary through return_as_dictionary
+        # method for comparison
+        test_dic = inventory.return_as_dictionary()
+
+        # Verify return_as_dictionary function populates
+        # Each item, and are properly instantiated
+        self.assertEqual(4, len(test_dic))
```
Comment (majstorpoki) [inline students/KyleCreek/lesson01/assignment/test_unit.py:56]: contains 4 components

File: students/KyleCreek/lesson01/assignment/test_unit.py [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+        self.assertEqual('1', test_dic['product_code'])
```
Comment (majstorpoki) [inline students/KyleCreek/lesson01/assignment/test_unit.py:57]: Test component names and values

File: students/KyleCreek/lesson01/assignment/test_unit.py [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+        Tests the return as dictionary method of
+        the inventory class. Ensures the length
+        of returned dictionary matches desired
+        length.
+
+        :return: None
+
+        """
+        # Create an instance of the Inventory Class
+        inventory = Inventory("1", "2", "3", "4")
```
Comment (majstorpoki) [inline students/KyleCreek/lesson01/assignment/test_unit.py:49]: Works, but may not be sufficiently descriptive

File: students/KyleCreek/lesson02/assignment/src/charges_calc.py [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+import math
+import logging
+
+
+# Create an input to set a logging level.
+
+
+def set_up_logger(debug):
+    """
+    Function to set up logger
```
Comment (majstorpoki) [inline students/KyleCreek/lesson02/assignment/src/charges_calc.py:19]: I like how you created a logger initializer

File: students/KyleCreek/lesson02/assignment/src/charges_calc.py [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+    :return:
+    """
+    # Change Debugger Level to an Integer
+    debug = int(debug)
```
Comment (majstorpoki) [inline students/KyleCreek/lesson02/assignment/src/charges_calc.py:23]: Why?

**Comment by KyleCreek** [comment]
Adding new lesson Files

**Comment by KyleCreek** [comment]
Trying to add lesson 04 Assignments due

**Comment by KyleCreek** [comment]
Trying to Add lesson04 Assignments due

File: students/KyleCreek/lesson03/assignment/src/basic_operations.py [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+    try:
+        # Query Customer Database for row
+        data = Customer.get(Customer.customer_id == customer_id)
+        # Place Data into a return dictionary
+        return_dict = {"First Name": data.first_name, "Last Name": data.last_name,
+                       "Email": data.email_address,"Phone Number": data.phone_number}
+        logging.info("The Following Dictionary was created: {}".format(return_dict))
+        return return_dict
+    # Exception where query target does not exist
+    except Exception as e:
```
Comment (majstorpoki) [inline students/KyleCreek/lesson03/assignment/src/basic_operations.py:31]: If we cannot find a record, is it really an exception? Maybe simple the record is not there...

File: students/KyleCreek/lesson03/assignment/src/basic_operations.py [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+                last_name=last_name,
+                home_address=home_address,
+                phone_number=phone_number,
+                email_address=email_address,
+                status=status,
+                credit_limit=credit_limit
+            )
+            new_customer.save()
+            logging.info("New Customer {} Added!".format(customer_id))
+    except peewee.IntegrityError:
```
Comment (majstorpoki) [inline students/KyleCreek/lesson03/assignment/src/basic_operations.py:74]: I like this

File: students/KyleCreek/lesson03/assignment/src/basic_operations.py [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+def update_customer_credit(customer_id, credit_limit):
+    """
+    Searches for an existing Customer by customer_id and
+    updates customer's credit limit.
+    :param customer_id: Customer's ID
+    :param credit_limit: Customers Credit Limit
+    :return: None
+    """
+    try:
+        Customer.update(credit_limit=credit_limit)\
```
Comment (majstorpoki) [inline students/KyleCreek/lesson03/assignment/src/basic_operations.py:103]: I like this use of update method

**Review by majstorpoki — APPROVED** [review]
Nice work!

**Review by ObjectiveJoe — APPROVED** [review]

**Pull request merged**